### DISCUSSION TITLE
Cannot find module 'postcss-preset-env' error when trying to build a CRA app

### DISCUSSION THREAD

**Discussion by runoncedev** [vercel/turborepo#807] [Help]
I'm trying to build a `create-react-app` app but I'm getting this error:

> Loading PostCSS "postcss-preset-env" plugin failed: Cannot find module 'postcss-preset-env'

If I run `npm install` inside the CRA app, it seems to work but I'm not sure if that's how you are supposed to install things inside the monorepo.

Here is an example: https://github.com/oncet/turborepo-cra

**Comment by nareshbhatia** [comment]
I am running into the exact same problem. I know that we are supposed to run `npm install` only from the root folder, but when I do that it seems to right away create a `node_modules` folder in the CRA app's folder. That doesn't look right - node_modules` should only be at the root level, right? So something in a CRA app is not working very well with workspaces.

**Reply by anthonyshew** [reply]
Just want to drop a quick note here for any future searchers:

Depending on your package manager and configuration, you may see different combinations of `node_modules` in the root and/or packages of your workspace. Don't worry if you see things installed where you weren't expecting! As long as they are (usually) resolving correctly, your package manager is working as intended.

**Comment by nareshbhatia** [comment]
Interesting enough, the issue only occurs with npm install. If I run yarn install, the issue goes away!!! So this may be an npm/CRA specific problem.

**Comment by YoselineGuerrero** [comment]
Just ran into this issue, I got no clue what's going on but I did create an app a few days ago and everything was fine. Now I get the same issue you guys got...

**Comment by eithan-hernandez** [comment]
i had the same error, try to check node and npm versions, i found out the lastest react version (installed when i run npx create-react-app) works with the node's lastest version. 
i used to work with node v14 then i changed it to node v17

**Reply by YoselineGuerrero** [reply]
Thanks! I had a feeling I needed to update my stuff, just got everything updated and no more problems!

**Reply by faizansayyed** [reply]
stuck in same issue, but getting this on build phase

#15 1.458    ▲ Next.js 14.1.0
#15 1.459 
#15 1.470    Creating an optimized production build ...
#15 14.15 Failed to compile.
#15 14.15 
#15 14.15 app/layout.tsx
#15 14.15 An error occured in `next/font`.
#15 14.15 
#15 14.15 Error: Cannot find module 'postcss-preset-mantine'
#15 14.15 Require stack:
#15 14.15 - /app/node_modules/next/dist/build/webpack/config/blocks/css/plugins.js
#15 14.15 - /app/node_modules/next/dist/build/webpack/config/blocks/css/index.js
#15 14.15 - /app/node_modules/next/dist/build/webpack/config/index.js
#15 14.15 - /app/node_modules/next/dist/build/webpack-config.js
#15 14.15 - /app/node_modules/next/dist/build/webpack-build/impl.js
#15 14.15 - /app/node_modules/next/dist/compiled/jest-worker/processChild.js
#15 14.15     at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
#15 14.15     at /app/node_modules/next/dist/server/require-hook.js:55:36
#15 14.15     at Function.resolve (node:internal/modules/helpers:187:19)
#15 14.15     at loadPlugin

**Comment by Feridun12** [comment]
I have same problem as well with you guys when I try to create a react app I have no idea what is going on. I have done npm install as well thinking that will change anything but I guess not.

**Reply by YoselineGuerrero** [reply]
check your versions like @eithan-hernandez says (node --version or just node -v and npm -v) I just updated and I'm at node v18.2.0 and npm 8.11.0 try to update and see if that will help you out.

**Reply by Feridun12** [reply]
Just updated everything but I still get the same problem. I updated npm to 8.11.0 but my node gets stuck at v14.17.6. It says it installs v18.2.0 but it gets stuck at v14.17.6. Any idea about this ?

**Reply by YoselineGuerrero** [reply]
Not sure what you're using (mac, windows, linux) nor how you are updating, can I get some more details? How are you updating node?

**Comment by AndrewCathcart** [comment]
did you folks ever resolve this?

**Comment by 03gabriel-coelho** [comment]
I had the same problem using "npx create-next-app@latest", and I solved it by installing "postcss-preset-env" like this:

https://github.com/csstools/postcss-plugins/blob/HEAD/plugin-packs/postcss-preset-env/INSTALL.md#nextjs

**Comment by RobinJiao** [comment]
npm install --save-dev postcss-preset-mantine

solved my issue
https://mantine.dev/styles/postcss-preset/
